pilot/pkg/xds/endpoint_builder.go

```diff
@@ -211,9 +212,9 @@ type LocalityEndpoints struct {
 	llbEndpoints endpoint.LocalityLbEndpoints
 }
 
-func (e *LocalityEndpoints) append(ep *model.IstioEndpoint, le *endpoint.LbEndpoint) {
+func (e *LocalityEndpoints) append(ep *model.IstioEndpoint, le ...*endpoint.LbEndpoint) {
```

Comment:
We have

```
func (e *LocalityEndpoints) AssertInvarianceInTest() {
	if len(e.llbEndpoints.LbEndpoints) != len(e.istioEndpoints) {
		panic(" len(e.llbEndpoints.LbEndpoints) != len(e.tunnelMetadata)")
	}
}
```

so this seems to violate that

Reply:
we need to update this assert to exclude waypoint case.

BTW, not sure why the integrate test failed